test/e2e/clusterctl_upgrade.go

```diff
@@ -201,21 +204,19 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		initKubernetesVersion   string
 
 		workloadClusterName string
-
-		scheme *apiruntime.Scheme
 	)
 
 	BeforeEach(func() {
-		scheme = apiruntime.NewScheme()
-		framework.TryAddDefaultSchemes(scheme)
-
 		Expect(ctx).NotTo(BeNil(), "ctx is required for %s spec", specName)
 		input = inputGetter()
 		Expect(input.E2EConfig).ToNot(BeNil(), "Invalid argument. input.E2EConfig can't be nil when calling %s spec", specName)
 		Expect(input.ClusterctlConfigPath).To(BeAnExistingFile(), "Invalid argument. input.ClusterctlConfigPath must be an existing file when calling %s spec", specName)
 		Expect(input.BootstrapClusterProxy).ToNot(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
 		Expect(input.InitWithBinary).ToNot(BeEmpty(), "Invalid argument. input.InitWithBinary can't be empty when calling %s spec", specName)
 		Expect(input.InitWithKubernetesVersion).ToNot(BeEmpty(), "Invalid argument. input.InitWithKubernetesVersion can't be empty when calling %s spec", specName)
+		if input.UseKindForManagementCluster {
+			Expect(input.KindManagementClusterNewClusterProxyFunc).ToNot(BeNil(), "Invalid argument. input.KindManagementClusterNewClusterProxyFunc can't be nil when calling %s spec", specName)
```

Comment:
I would prefer defaulting KindManagementClusterNewClusterProxyFunc instead of forcing everyone who sets UseKindForManagementCluster to explicitly set it